staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -38,8 +39,8 @@ func findServicePort(svc *v1.Service, port int32) (*v1.ServicePort, error) {
 	return nil, errors.NewServiceUnavailable(fmt.Sprintf("no service port %d found for service %q", port, svc.Name))
 }
 
-// ResourceLocation returns a URL to which one can send traffic for the specified service.
-func ResolveEndpoint(services listersv1.ServiceLister, endpoints listersv1.EndpointsLister, namespace, id string, port int32) (*url.URL, error) {
+// ResolveEndpoint returns a URL to which one can send traffic for the specified service.
+func ResolveEndpoint(services listersv1.ServiceLister, endpoints listersv1.EndpointsLister, namespace, id string, port int32, seenEndpoints ...*url.URL) (*url.URL, error) {
```

Comment:
doc that the seen endpoints are not returned. One could call it more clearly: avoidEndpoints, excludedEndpoints